compiler/src/iree/compiler/Modules/IO/Parameters/Transforms/GenerateSplatParameterArchive.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 The IREE Authors
```

Comment:
That's what I'm doing in here: https://github.com/openxla/iree/pull/16353/files#diff-ab93fa640f3af87006a34823de4567a329e26177c2a869185959a604c2676d44

I just put it e2e because I figured the `iree-dump-parameters` tool was already included there and I shouldn't include it in a compiler lit test. Also this pass does not modify the IR at all so I didn't think it was worth checking the MLIR.

Reply:
The e2e test uses `iree-run-module`, which makes it an "integration test" (involves both the compiler and the runtime). A compiler lit test using `iree-compile`/`iree-opt` and `iree-dump-parameters` I'd consider to be a "unit test".

As part of the "package CI" refactoring, I've been moving "unit tests" to run right after the build step on CPU machines while "integration tests" will run in following steps on test CPU/GPU/etc. machines. Any test that only uses compiler tools on the host should aim to get into that first "unit test" category.

(various literature differentiates between "end-to-end" testing and "integration" testing, but I haven't really been making a distinction on this project)

Reply:
I see, makes sense to me. I can make the parameter check a lit test then.

Reply:
yep! having iree-dump-parameters required for the lit unit test seems fine to me (we try to shy away from any big deps in there that require frequent rebuilding during iteration to run tests, like if we required clang to be built to run any simple compiler tests - but iree-dump-parameters is like 8 files to compile and doesn't change when the compiler does so it's ok to build infrequently)